controller/users.go

```diff
@@ -1061,6 +1062,13 @@ func filterUsers(appl application.Application, ctx *app.ListUsersContext) ([]acc
 		// .. Add other filters in future when needed into the userFilters slice in the above manner.
 		if len(userFilters) != 0 {
 			filteredUsers, err = appl.Users().Query(userFilters...)
+			if err != nil {
+				return nil, nil, errs.Wrap(err, "error fetching users")
+			}
+			// Should be zero or the only user in the result. If the user's email is private then ignore the user
+			if len(filteredUsers) > 0 && filteredUsers[0].EmailPrivate {
```

Comment:
I think, it might be a lot cleaner if we introduced a 
```
account.UserFilterByEmailPrivacy(showPrivateEmails bool)
```
instead of adding custom logic. Do you think its worth it?

Reply:
Yes. I agree. The code would be cleaner. Will add it.

Reply:
Done - https://github.com/fabric8-services/fabric8-auth/pull/393/commits/a95d5821603eb9901efe189e6b586d69d3a90654